modules/gdscript/gdscript_parser.cpp

```diff
@@ -4224,17 +4308,80 @@ bool GDScriptParser::export_group_annotations(const AnnotationNode *p_annotation
 
 bool GDScriptParser::warning_annotations(const AnnotationNode *p_annotation, Node *p_target, ClassNode *p_class) {
 #ifdef DEBUG_ENABLED
```

Comment:
What I meant [here](https://github.com/godotengine/godot/pull/76020#discussion_r1190040033) is to refactor the `#ifdef` into a `#ifndef`.

```suggestion
bool GDScriptParser::warning_annotations(const AnnotationNode *p_annotation, Node *p_target, ClassNode *p_class) {
#ifndef DEBUG_ENABLED
	// Only available in debug builds.
	return true;
#endif
```

Reply:
It probably makes sense to handle the simple case first, but isn't it "don't begin with a not" like above? Also, the return values of annotation application methods are not used, we could remove them in a separate PR.

Reply:
It's more akin the [Return early pattern](https://medium.com/swlh/return-early-pattern-3d18a41bba8) than anything else.